torch/csrc/api/include/torch/nn/functional/vision.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <torch/types.h>
+
+namespace torch {
+namespace nn {
+namespace functional {
+
+inline Tensor affine_grid(
+    const Tensor& theta,
+    const IntArrayRef& size,
+    bool align_corners = false) {
+  TORCH_CHECK(
+      theta.is_floating_point(), "Expected theta to have floating point type.");
+
+  if (size.size() == 4) {
+    TORCH_CHECK(
+        theta.dim() == 3 && theta.size(1) == 2 && theta.size(2) == 3,
```

Comment:
It would be great to check if the following works:
```suggestion
        theta.dim() == 3 && theta.size(-2) == 2 && theta.size(-1) == 3,
```
to match the Python version even better.